src/renderer/utils/InstallManager.tsx

```diff
@@ -688,6 +594,167 @@ export class InstallManager {
+    const installedTrack = store.getState().installedTracks[addon.key] as AddonTrack;
+
+    if (installedTrack) {
+      return installedTrack;
+    }
+
+    const install = this.getAddonInstall(Directories.inInstallLocation(addon.targetDirectory));
+
+    if (!install) {
+      return null;
+    }
+
+    const matchingTrack = addon.tracks.find((it) => it.url === install.source);
+
+    if (!matchingTrack) {
+      return null;
+    }
+
+    this.setCurrentlyInstalledTrack(addon, matchingTrack);
+
+    return matchingTrack;
+  }
+
+  private static lowestAvailableAbortControllerID(): number {
+    for (let i = 0; i < this.abortControllers.length; i++) {
+      if (
+        !store
+          .getState()
+          .downloads.map((download) => download.abortControllerID)
+          .includes(i)
+      ) {
+        return i;
+      }
+    }
+  }
+
+  private static async determineAddonInstallStatus(addon: Addon): Promise<InstallState> {
+    console.log('Checking install status');
+
+    const installDir = Directories.inInstallLocation(addon.targetDirectory);
+
+    console.log('Checking for git install');
+
+    if (Directories.isGitInstall(installDir)) {
+      return { status: InstallStatus.GitInstall };
+    }
+
+    const addonInstalledTrack = this.determineAddonInstalledTrack(addon);
+    const addonSelectedTrack = this.getAddonSelectedTrack(addon);
+
+    if (!addonInstalledTrack) {
+      return { status: InstallStatus.NotInstalled };
+    }
+
+    try {
+      const updateInfo = await new FragmenterUpdateChecker().needsUpdate(addonSelectedTrack.url, installDir, {
+        forceCacheBust: true,
+      });
+
+      console.log('Update info', updateInfo);
+
+      if (addonSelectedTrack !== addonInstalledTrack && addonInstalledTrack) {
+        return { status: InstallStatus.TrackSwitch };
+      }
+
+      if (updateInfo.isFreshInstall) {
+        return { status: InstallStatus.NotInstalled };
+      }
+
+      if (updateInfo.needsUpdate) {
+        return { status: InstallStatus.NeedsUpdate };
+      }
+
+      return { status: InstallStatus.UpToDate };
+    } catch (e) {
+      console.error(e);
+      return { status: InstallStatus.Unknown };
+    }
+  }
+
+  public static async checkForUpdates(addon: Addon): Promise<void> {
+    console.log('Checking for updates for ' + addon.key);
+
+    const installDir = Directories.inInstallLocation(addon.targetDirectory);
+
+    const state = store.getState();
+
+    const fragmenterUpdateChecker = new FragmenterUpdateChecker();
+
+    for (const track of addon.tracks) {
+      const updateInfo = await fragmenterUpdateChecker.needsUpdate(track.url, installDir, { forceCacheBust: true });
+
+      let info: ReleaseInfo;
+      if (track.releaseModel.type === 'fragmenter') {
+        info = {
+          name: updateInfo.distributionManifest.version,
+          changelogUrl: undefined,
+          releaseDate: Date.now(),
+        };
```

Comment:
Can the missing properties be checked for existence instead and included in the fragmenter distributionManifest wherever possible?

Reply:
The properties currently never exist on the distribution manifest.

Reply:
Can we make them existent? Shouldn't be too hard.